dockstore-integration-testing/src/test/java/io/dockstore/webservice/ZenodoIT.java

```diff
@@ -0,0 +1,381 @@
+
+        // Release a branch. Should not automatically create a DOI because it's not a tag
+        handleGitHubRelease(workflowsApi, DockstoreTesting.WORKFLOW_DOCKSTORE_YML, "refs/heads/master", USER_2_USERNAME);
+        foobar2 = workflowsApi.getWorkflow(foobar2Id, "versions");
+        WorkflowVersion foobar2BranchVersion = getWorkflowVersion(foobar2, "master").orElse(null);
+        assertNotNull(foobar2BranchVersion);
+        assertFalse(foobar2BranchVersion.getDois().containsKey(DoiCreator.DOCKSTORE.name()));
+    }
+
+    @Test
+    void testRefreshAutomaticDoiCreation(Hoverfly hoverfly) {
+        hoverfly.simulate(ZENODO_SIMULATION_SOURCE);
+        ApiClient webClient = getOpenAPIWebClient(true, USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsApi = new WorkflowsApi(webClient);
+
+        // register workflow and refresh workflow
+        Workflow workflow = workflowsApi
+                .manualRegister(SourceControl.GITHUB.name(), DockstoreTesting.HELLO_WDL_WORKFLOW, "/Dockstore.wdl", "", DescriptorType.WDL.toString(), "/test.json");
+        workflowsApi.refresh1(workflow.getId(), false);
+        workflow = workflowsApi.getWorkflow(workflow.getId(), "versions");
+
+        // Get a valid tag
+        WorkflowVersion tagVersion = getWorkflowVersion(workflow, "1.1").orElse(null);
+        assertTrue(tagVersion.getDois().isEmpty(), "Should not have any automatic DOIs because it's unpublished");
+
+        // Publish the workflow and refresh
+        workflowsApi.publish1(workflow.getId(), CommonTestUtilities.createOpenAPIPublishRequest(true));
+        workflowsApi.refresh1(workflow.getId(), false);
+
+        tagVersion = workflowsApi.getWorkflowVersionById(workflow.getId(), tagVersion.getId(), "");
+        assertNotNull(tagVersion.getDois().get(DoiCreator.DOCKSTORE.name()).getName(), "Should have automatic DOI because it's a valid published tag");
+    }
+
+    @Test
+    void testAutomaticDoiCreationOnPublishGitHubApp(Hoverfly hoverfly) {
+        hoverfly.simulate(ZENODO_SIMULATION_SOURCE);
+        ApiClient webClient = getOpenAPIWebClient(true, USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsApi = new WorkflowsApi(webClient);
+
+        // Create a GitHub App workflow
+        handleGitHubRelease(workflowsApi, DockstoreTesting.WORKFLOW_DOCKSTORE_YML, "refs/tags/0.8", USER_2_USERNAME);
+        Workflow workflow = workflowsApi.getWorkflowByPath("github.com/" + DockstoreTesting.WORKFLOW_DOCKSTORE_YML + "/foobar2", WorkflowSubClass.BIOWORKFLOW, "versions");
+        WorkflowVersion tagVersion = getWorkflowVersion(workflow, "0.8").orElse(null);
+        assertNotNull(tagVersion);
+        // No DOIs should've been automatically created because the workflow is unpublished
+        assertTrue(workflow.getConceptDois().isEmpty());
+        assertTrue(tagVersion.getDois().isEmpty());
+
+        // Publish workflow, should automatically create DOIs
```

Comment:
How many versions does this go back? 
Will likely be answered later

Reply:
On publish, it will look at all versions https://github.com/dockstore/dockstore/blob/7d1b34a1c9b9a152acd27e4387461c9736a41328/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L760